### PR TITLE
Can now run multiple OscarMTProducers in the same job

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

This was needed to test generating GEN+SIM in mixing module to generate pileup on the fly. The problem is Geant4 requires that it be initialized on one and only one thread which means all instances of OscarMTProducer must share that thread.

#### PR validation:

Code compiles. Test of on-the-fly pileup succeeded.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: FWCore/Concurrency/interface/include_first_syncWait.h [hunk 1]
```diff
@@ -8,32 +8,19 @@
...
-          if (iExcept) {
-            exceptPtr = *iExcept;
-          }
-          oneapi::tbb::task::resume(tag);
-        });
-        iFunc(WaitingTaskHolder(group, waitTask));
-      });  //suspend
-    });    //group.run
+    FinalWaitingTask last{group};
+    group.run([&]() { iFunc(WaitingTaskHolder(group, &last)); });  //group.run
```
Comment (makortel) [inline FWCore/Concurrency/interface/include_first_syncWait.h:21]: This PR appears to include the contents of #40253. Could these be removed? Then the `core` signature would not be needed.

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by Dr15Jones** [comment]
please test

**Comment by makortel** [comment]
Comparison differences are in 11634.7 and thus spurious (#39803).

**Comment by civanch** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**